### PR TITLE
chore(Messaggi a valore legale): [IAMVL-68] Preview MVL attachments on IOS

### PR DESCRIPTION
**Pull request by pietro909**
## Short description
We implement the flow to let the user preview a PDF attachment in-app. It only supports Android at the moment and **is behind feature-flag**.

## List of changes proposed in this pull request
- add [react-native-pdf](https://www.npmjs.com/package/react-native-pdf)
- implement [a new component](https://github.com/pagopa/io-app/pull/3774/files#diff-08db2f5b09754859765cbf0bf32242d6eb0c46d0ca027324abc02b7939ca9fe2R66) to present the PDF in a modal (no test for now as we are still experimenting with this feature and might be dropped)
- update [handleDownloadResult](https://github.com/pagopa/io-app/pull/3774/files#diff-03e64fd8c3f3946d74994448b3bfe5488063b5072f85056cc9c91601e0c6deb4R52) to inject the new behavior

## How to test

1. configure `io-dev-server` to return MVLs by adding `messages: { legalCount: 2 }` in [config/config.json](https://github.com/pagopa/io-dev-api-server/blob/master/config/config.json)
2. enable both `PAGINATED_MESSAGES` and `MVL_ENABLED` [feature flags](https://github.com/pagopa/io-app/blob/master/.env.local#L57) in your `.env` file
3. run on IOS
4. open an MVL
5. tap on the PDF attachment
6. notice how it opens the preview
7. try with an image
8. notice how it opens the native menu

https://user-images.githubusercontent.com/2094604/154714381-637694bc-adb9-4fa9-964e-d5301bd6bd67.mov

### CODE REVIEW

File: ts/features/mvl/screens/details/components/attachment/DownloadAttachmentConfirmationBottomSheet.tsx
Comment (dgopsq) [inline ts/features/mvl/screens/details/components/attachment/DownloadAttachmentConfirmationBottomSheet.tsx:222]: **⚠️ Opinionated comment incoming**

Since the `onConfirm` prop is becoming quite big, what do you think about separating _at least_ the `showPreview` callback of `handleDownloadResult` into a separate function? Something like:

```ts
const handleShowPreview = (path: string, actionConfig: ActionConfig) =>
    showAnimatedModal(
      <PdfPreview
        path={path}
        onClose={hideModal}
        onError={_error => {
          dismiss();
          showToast(
            i18n.t(
              "features.mvl.details.attachments.bottomSheet.failing.details"
            )
          );
        }}
        actionConfig={actionConfig}
      />,
      BottomTopAnimation
    );
```

**Review by dgopsq — CHANGES_REQUESTED** [review]
I left some comment here and there, imho the implementation you developed passing platform-specific callbacks is ok, I don't think it's possible to do something more with the current bottom sheets implementation!

Also, I don't really know if it's something already known but there is a bottom sheet flash after ignoring the warning using the checkbox:

https://user-images.githubusercontent.com/5963683/154992519-191c00da-7ce3-4932-aa53-2073c53106c9.mov

**Comment by pietro909** [comment]
Thank you for pointing out the issue with the loader @dgopsq ! I've been testing with an arbitrary delay on that endpoint to ensure the loader works as expected and missed the case when it's faster 🤦 

I've added [a comment](https://pagopa.atlassian.net/browse/IAMVL-68?focusedCommentId=32303) to the task to ensure we address the issue via UI/UX update rather than patching it up here.

File: ts/features/mvl/screens/details/components/attachment/DownloadAttachmentConfirmationBottomSheet.tsx
Comment (pietro909) [reply ts/features/mvl/screens/details/components/attachment/DownloadAttachmentConfirmationBottomSheet.tsx:222]: It's a fair point, yet given that this part will likely be reworked soon I'd rather keep it as it is.
I'll make sure to address this concern once we get the big picture 👍

File: ts/features/mvl/screens/details/components/attachment/PdfPreview.tsx
Comment (pietro909) [reply ts/features/mvl/screens/details/components/attachment/PdfPreview.tsx]: Wow, how could've missed it?! Thanks, fixed [here](https://github.com/pagopa/io-app/pull/3774/commits/e1f2d6a68bb7ebbcbb05c06c035b0e8f9a72b561#diff-08db2f5b09754859765cbf0bf32242d6eb0c46d0ca027324abc02b7939ca9fe2R18)

File: ts/features/mvl/screens/details/components/attachment/PdfPreview.tsx
Comment (pietro909) [reply ts/features/mvl/screens/details/components/attachment/PdfPreview.tsx]: Definitely :-)  [here you go](https://github.com/pagopa/io-app/pull/3774/commits/e1f2d6a68bb7ebbcbb05c06c035b0e8f9a72b561#diff-08db2f5b09754859765cbf0bf32242d6eb0c46d0ca027324abc02b7939ca9fe2R80

File: ts/features/mvl/utils.ts
Comment (pietro909) [reply ts/features/mvl/utils.ts]: Makes sense - [renamed](https://github.com/pagopa/io-app/pull/3774/commits/e1f2d6a68bb7ebbcbb05c06c035b0e8f9a72b561#diff-03e64fd8c3f3946d74994448b3bfe5488063b5072f85056cc9c91601e0c6deb4R32)

**Review by dgopsq — APPROVED** [review]
LGTM, @thisisjp waiting for your approval too 🙂

**Review by thisisjp — APPROVED** [review]
okay for me, I also posted a suggestion on the related JIRA task created by @pietro909

**Pull request merged**
